source/particle/property/initial_composition.cc

```diff
@@ -53,7 +44,7 @@ namespace aspect
         for (unsigned int i = 0; i < this->n_compositional_fields(); i++)
           {
             std::ostringstream field_name;
-            field_name << "initial C_" << i;
+            field_name << "initial composition_" << i;
```

Comment:
We are using these names for querying properties by name, by propagating what is returned here into the array ultimately used by `Manager<dim>::get_property_component_by_name()`. It seems to me that this function is not actually used anywhere right now, so here's an idea for a non-backward compatible change: instead of using the generic `composition_{$i}` text, could we instead query the compositional field for its name itself?